script/deployment/bash/common_functions.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+set -e
+
+# Read environment variables
+source .env
+
+# Get CHAIN_ID automatically using RPC_URL
+CHAIN_ID=$(cast chain-id --rpc-url $RPC_URL)
+echo "Detected Chain ID: $CHAIN_ID"
+
+# Set config and output directories
+CONFIG_DIR="script/config/$CHAIN_ID/42420"
+CONFIG_FILE="$CONFIG_DIR/targetContractSetConfig.json"
+OUTPUT_FILE="$CONFIG_DIR/targetContractAddresses.json"
```

Comment:
move relevant sections to proper functions

Reply:
this is not part of this story... I merge from develop and probably got some squashed feature